### ISSUE TITLE
Container with macvlan IP uses wrong IP after recreation

### ISSUE THREAD

**Issue by ptr727** [containrrr/watchtower#1912]
### Describe the bug

I get an alert from my Unifi network controller that I have a duplicate IP on my network.  
The duplicate originates from my HA container, that was created to use 192.168.26 as IP, but is now using 192.168.1.3, that is a duplicate.  
Today is the 2nd time I observed this, inspecting the HA container shows it is again using 192.168.1.3 instead of 192.168.1.26 as when created.

This may be related to #1906 that is related to a docker update, as I've noticed it about the same time docker was updated to v25, but could be coincidence.

### Steps to reproduce

1. Create container using macvlan and static IP.
2. Watchtower updates container.
3. Container is now using the wrong IP.

### Expected behavior

Updates use the same IP.

### Screenshots

<img width="489" alt="image" src="https://github.com/containrrr/watchtower/assets/2061579/23ec5994-c042-4122-81af-61bed8ea69a7">

<img width="760" alt="image" src="https://github.com/containrrr/watchtower/assets/2061579/3d247e3a-c30a-4156-9e08-959e1598d338">

### Environment

- Platform: Proxmox / Debian
- Architecture: x64
- Docker Version: 25.0.1, build 29cf629

### Your logs

```text
TODO: Run with debug, normal logs showed nothing interesting
```

### Additional context

Freshly created container:

```json
{
    "Id": "cb17b2321ee3a0f92032bbad8d2209642a46dc70858c7c8cf45e043ccf34278a",
    "Created": "2024-01-27T15:17:17.556091692Z",
    "Path": "/init",
    "Args": [],
    "State": {
        "Status": "running",
        "Running": true,
        "Paused": false,
        "Restarting": false,
        "OOMKilled": false,
        "Dead": false,
        "Pid": 3336641,
        "ExitCode": 0,
        "Error": "",
        "StartedAt": "2024-01-27T15:17:18.672315028Z",
        "FinishedAt": "0001-01-01T00:00:00Z"
    },
    "Image": "sha256:ec63b25683486d7aaccc90e078d4e3853d36f7ec3578e2560c9e3372c59f117b",
    "ResolvConfPath": "/var/lib/docker/containers/cb17b2321ee3a0f92032bbad8d2209642a46dc70858c7c8cf45e043ccf34278a/resolv.conf",
    "HostnamePath": "/var/lib/docker/containers/cb17b2321ee3a0f92032bbad8d2209642a46dc70858c7c8cf45e043ccf34278a/hostname",
    "HostsPath": "/var/lib/docker/containers/cb17b2321ee3a0f92032bbad8d2209642a46dc70858c7c8cf45e043ccf34278a/hosts",
    "LogPath": "/var/lib/docker/containers/cb17b2321ee3a0f92032bbad8d2209642a46dc70858c7c8cf45e043ccf34278a/cb17b2321ee3a0f92032bbad8d2209642a46dc70858c7c8cf45e043ccf34278a-json.log",
    "Name": "/home-assistant",
    "RestartCount": 0,
    "Driver": "zfs",
    "Platform": "linux",
    "MountLabel": "",
    "ProcessLabel": "",
    "AppArmorProfile": "docker-default",
    "ExecIDs": null,
    "HostConfig": {
        "Binds": [
            "/data/appdata/home-assistant/config:/config:rw"
        ],
        "ContainerIDFile": "",
        "LogConfig": {
            "Type": "json-file",
            "Config": {}
        },
        "NetworkMode": "aaa_publicnet",
        "PortBindings": null,
        "RestartPolicy": {
            "Name": "unless-stopped",
            "MaximumRetryCount": 0
        },
        "AutoRemove": false,
        "VolumeDriver": "",
        "VolumesFrom": null,
        "ConsoleSize": [
            0,
            0
        ],
        "CapAdd": null,
        "CapDrop": null,
        "CgroupnsMode": "private",
        "Dns": null,
        "DnsOptions": null,
        "DnsSearch": null,
        "ExtraHosts": null,
        "GroupAdd": null,
        "IpcMode": "private",
        "Cgroup": "",
        "Links": null,
        "OomScoreAdj": 0,
        "PidMode": "",
        "Privileged": false,
        "PublishAllPorts": false,
        "ReadonlyRootfs": false,
        "SecurityOpt": null,
        "UTSMode": "",
        "UsernsMode": "",
        "ShmSize": 67108864,
        "Runtime": "runc",
        "Isolation": "",
        "CpuShares": 0,
        "Memory": 0,
        "NanoCpus": 0,
        "CgroupParent": "",
        "BlkioWeight": 0,
        "BlkioWeightDevice": null,
        "BlkioDeviceReadBps": null,
        "BlkioDeviceWriteBps": null,
        "BlkioDeviceReadIOps": null,
        "BlkioDeviceWriteIOps": null,
        "CpuPeriod": 0,
        "CpuQuota": 0,
        "CpuRealtimePeriod": 0,
        "CpuRealtimeRuntime": 0,
        "CpusetCpus": "",
        "CpusetMems": "",
        "Devices": null,
        "DeviceCgroupRules": null,
        "DeviceRequests": null,
        "MemoryReservation": 0,
        "MemorySwap": 0,
        "MemorySwappiness": null,
        "OomKillDisable": null,
        "PidsLimit": null,
        "Ulimits": null,
        "CpuCount": 0,
        "CpuPercent": 0,
        "IOMaximumIOps": 0,
        "IOMaximumBandwidth": 0,
        "MaskedPaths": [
            "/proc/asound",
            "/proc/acpi",
            "/proc/kcore",
            "/proc/keys",
            "/proc/latency_stats",
            "/proc/timer_list",
            "/proc/timer_stats",
            "/proc/sched_debug",
            "/proc/scsi",
            "/sys/firmware",
            "/sys/devices/virtual/powercap"
        ],
        "ReadonlyPaths": [
            "/proc/bus",
            "/proc/fs",
            "/proc/irq",
            "/proc/sys",
            "/proc/sysrq-trigger"
        ]
    },
    "GraphDriver": {
        "Data": {
            "Dataset": "ssdpool/docker/c1979b4eeffa24dd2b5ebaba12289702b28e4ece7326a5809207c6e0055f1e7e",
            "Mountpoint": "/var/lib/docker/zfs/graph/c1979b4eeffa24dd2b5ebaba12289702b28e4ece7326a5809207c6e0055f1e7e"
        },
        "Name": "zfs"
    },
    "Mounts": [
        {
            "Type": "bind",
            "Source": "/data/appdata/home-assistant/config",
            "Destination": "/config",
            "Mode": "rw",
            "RW": true,
            "Propagation": "rprivate"
        }
    ],
    "Config": {
        "Hostname": "home-assistant",
        "Domainname": "home.insanegenius.net",
        "User": "",
        "AttachStdin": false,
        "AttachStdout": true,
        "AttachStderr": true,
        "Tty": false,
        "OpenStdin": false,
        "StdinOnce": false,
        "Env": [
            "TZ=America/Los_Angeles",
            "PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "LANG=C.UTF-8",
            "S6_BEHAVIOUR_IF_STAGE2_FAILS=2",
            "S6_CMD_WAIT_FOR_SERVICES_MAXTIME=0",
            "S6_CMD_WAIT_FOR_SERVICES=1",
            "S6_SERVICES_READYTIME=50",
            "S6_SERVICES_GRACETIME=240000"
        ],
        "Cmd": null,
        "Image": "docker.io/homeassistant/home-assistant:latest",
        "Volumes": null,
        "WorkingDir": "/config",
        "Entrypoint": [
            "/init"
        ],
        "MacAddress": "02:42:c0:a8:01:1a",
        "OnBuild": null,
        "Labels": {
            "com.centurylinklabs.watchtower.depends-on": "zigbee2mqtt,mariadb",
            "com.centurylinklabs.watchtower.enable": "true",
            "io.hass.arch": "amd64",
            "io.hass.base.arch": "amd64",
            "io.hass.base.image": "ghcr.io/home-assistant/amd64-base:3.18",
            "io.hass.base.name": "python",
            "io.hass.base.version": "2023.10.4",
            "io.hass.type": "core",
            "io.hass.version": "2024.1.5",
            "org.opencontainers.image.authors": "The Home Assistant Authors",
            "org.opencontainers.image.created": "2024-01-20 19:03:51+00:00",
            "org.opencontainers.image.description": "Open-source home automation platform running on Python 3",
            "org.opencontainers.image.documentation": "https://www.home-assistant.io/docs/",
            "org.opencontainers.image.licenses": "Apache License 2.0",
            "org.opencontainers.image.source": "https://github.com/home-assistant/core",
            "org.opencontainers.image.title": "Home Assistant",
            "org.opencontainers.image.url": "https://www.home-assistant.io/",
            "org.opencontainers.image.version": "2024.1.5",
            "traefik.enable": "true",
            "traefik.http.routers.home-assistant.entrypoints": "websecure",
            "traefik.http.routers.home-assistant.rule": "Host(`home-assistant-web.home.insanegenius.net`)",
            "traefik.http.services.home-assistant.loadbalancer.server.port": "8123",
            "traefik.http.services.home-assistant.loadbalancer.server.scheme": "http"
        }
    },
    "NetworkSettings": {
        "Bridge": "",
        "SandboxID": "9cf487f151665fea16ff2ed61d02a6918e85215816430d098a649e9e91ca1084",
        "SandboxKey": "/var/run/docker/netns/9cf487f15166",
        "Ports": {},
        "HairpinMode": false,
        "LinkLocalIPv6Address": "",
        "LinkLocalIPv6PrefixLen": 0,
        "SecondaryIPAddresses": null,
        "SecondaryIPv6Addresses": null,
        "EndpointID": "",
        "Gateway": "",
        "GlobalIPv6Address": "",
        "GlobalIPv6PrefixLen": 0,
        "IPAddress": "",
        "IPPrefixLen": 0,
        "IPv6Gateway": "",
        "MacAddress": "",
        "Networks": {
            "aaa_publicnet": {
                "IPAMConfig": {
                    "IPv4Address": "192.168.1.26"
                },
                "Links": null,
                "Aliases": [
                    "cb17b2321ee3",
                    "home-assistant"
                ],
                "MacAddress": "02:42:c0:a8:01:1a",
                "NetworkID": "233f37ba541b01a98137b9ff90137c409ca1b7c544b669e4d064448a3b66ea37",
                "EndpointID": "c56a194786f41f7ca5acea62d8993a73b7cb0caee3c94aeb680ac4b28bdf43da",
                "Gateway": "192.168.1.1",
                "IPAddress": "192.168.1.26",
                "IPPrefixLen": 24,
                "IPv6Gateway": "",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "DriverOpts": null,
                "DNSNames": [
                    "home-assistant",
                    "cb17b2321ee3"
                ]
            },
            "localnet": {
                "IPAMConfig": null,
                "Links": null,
                "Aliases": [
                    "cb17b2321ee3",
                    "home-assistant"
                ],
                "MacAddress": "02:42:ac:12:00:09",
                "NetworkID": "742700da91d804fe03f5b793f35309def9ee34e7f7a2c815eaa303cc32d4364d",
                "EndpointID": "52fc767189af8bcd076ab9146cf15db115ffd5fbd6ec8856af90955a738e2472",
                "Gateway": "172.18.0.1",
                "IPAddress": "172.18.0.9",
                "IPPrefixLen": 16,
                "IPv6Gateway": "",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "DriverOpts": null,
                "DNSNames": [
                    "home-assistant",
                    "cb17b2321ee3"
                ]
            }
        }
    }
}
```

Recreated container:

```json
{
    "Id": "de6c479604e6efa126d2ff97731ec25a413c374f24f8c7a3806505d34f923c29",
    "Created": "2024-01-27T10:04:05.564672397Z",
    "Path": "/init",
    "Args": [],
    "State": {
        "Status": "exited",
        "Running": false,
        "Paused": false,
        "Restarting": false,
        "OOMKilled": false,
        "Dead": false,
        "Pid": 0,
        "ExitCode": 0,
        "Error": "",
        "StartedAt": "2024-01-27T10:04:06.641217806Z",
        "FinishedAt": "2024-01-27T15:07:53.165092Z"
    },
    "Image": "sha256:ec63b25683486d7aaccc90e078d4e3853d36f7ec3578e2560c9e3372c59f117b",
    "ResolvConfPath": "/var/lib/docker/containers/de6c479604e6efa126d2ff97731ec25a413c374f24f8c7a3806505d34f923c29/resolv.conf",
    "HostnamePath": "/var/lib/docker/containers/de6c479604e6efa126d2ff97731ec25a413c374f24f8c7a3806505d34f923c29/hostname",
    "HostsPath": "/var/lib/docker/containers/de6c479604e6efa126d2ff97731ec25a413c374f24f8c7a3806505d34f923c29/hosts",
    "LogPath": "/var/lib/docker/containers/de6c479604e6efa126d2ff97731ec25a413c374f24f8c7a3806505d34f923c29/de6c479604e6efa126d2ff97731ec25a413c374f24f8c7a3806505d34f923c29-json.log",
    "Name": "/home-assistant",
    "RestartCount": 0,
    "Driver": "zfs",
    "Platform": "linux",
    "MountLabel": "",
    "ProcessLabel": "",
    "AppArmorProfile": "docker-default",
    "ExecIDs": null,
    "HostConfig": {
        "Binds": [
            "/data/appdata/home-assistant/config:/config:rw"
        ],
        "ContainerIDFile": "",
        "LogConfig": {
            "Type": "json-file",
            "Config": {}
        },
        "NetworkMode": "aaa_publicnet",
        "PortBindings": null,
        "RestartPolicy": {
            "Name": "unless-stopped",
            "MaximumRetryCount": 0
        },
        "AutoRemove": false,
        "VolumeDriver": "",
        "VolumesFrom": null,
        "ConsoleSize": [
            0,
            0
        ],
        "CapAdd": null,
        "CapDrop": null,
        "CgroupnsMode": "private",
        "Dns": [],
        "DnsOptions": [],
        "DnsSearch": [],
        "ExtraHosts": null,
        "GroupAdd": null,
        "IpcMode": "private",
        "Cgroup": "",
        "Links": null,
        "OomScoreAdj": 0,
        "PidMode": "",
        "Privileged": false,
        "PublishAllPorts": false,
        "ReadonlyRootfs": false,
        "SecurityOpt": null,
        "UTSMode": "",
        "UsernsMode": "",
        "ShmSize": 67108864,
        "Runtime": "runc",
        "Isolation": "",
        "CpuShares": 0,
        "Memory": 0,
        "NanoCpus": 0,
        "CgroupParent": "",
        "BlkioWeight": 0,
        "BlkioWeightDevice": null,
        "BlkioDeviceReadBps": null,
        "BlkioDeviceWriteBps": null,
        "BlkioDeviceReadIOps": null,
        "BlkioDeviceWriteIOps": null,
        "CpuPeriod": 0,
        "CpuQuota": 0,
        "CpuRealtimePeriod": 0,
        "CpuRealtimeRuntime": 0,
        "CpusetCpus": "",
        "CpusetMems": "",
        "Devices": null,
        "DeviceCgroupRules": null,
        "DeviceRequests": null,
        "MemoryReservation": 0,
        "MemorySwap": 0,
        "MemorySwappiness": null,
        "OomKillDisable": null,
        "PidsLimit": null,
        "Ulimits": null,
        "CpuCount": 0,
        "CpuPercent": 0,
        "IOMaximumIOps": 0,
        "IOMaximumBandwidth": 0,
        "MaskedPaths": [
            "/proc/asound",
            "/proc/acpi",
            "/proc/kcore",
            "/proc/keys",
            "/proc/latency_stats",
            "/proc/timer_list",
            "/proc/timer_stats",
            "/proc/sched_debug",
            "/proc/scsi",
            "/sys/firmware"
        ],
        "ReadonlyPaths": [
            "/proc/bus",
            "/proc/fs",
            "/proc/irq",
            "/proc/sys",
            "/proc/sysrq-trigger"
        ]
    },
    "GraphDriver": {
        "Data": {
            "Dataset": "ssdpool/docker/dcb5a7d893d91cddbe91c57fa4d993f5798e24135daea27945785cb769079d00",
            "Mountpoint": "/var/lib/docker/zfs/graph/dcb5a7d893d91cddbe91c57fa4d993f5798e24135daea27945785cb769079d00"
        },
        "Name": "zfs"
    },
    "Mounts": [
        {
            "Type": "bind",
            "Source": "/data/appdata/home-assistant/config",
            "Destination": "/config",
            "Mode": "rw",
            "RW": true,
            "Propagation": "rprivate"
        }
    ],
    "Config": {
        "Hostname": "home-assistant",
        "Domainname": "home.insanegenius.net",
        "User": "",
        "AttachStdin": false,
        "AttachStdout": true,
        "AttachStderr": true,
        "Tty": false,
        "OpenStdin": false,
        "StdinOnce": false,
        "Env": [
            "TZ=America/Los_Angeles",
            "PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "LANG=C.UTF-8",
            "S6_BEHAVIOUR_IF_STAGE2_FAILS=2",
            "S6_CMD_WAIT_FOR_SERVICES_MAXTIME=0",
            "S6_CMD_WAIT_FOR_SERVICES=1",
            "S6_SERVICES_READYTIME=50",
            "S6_SERVICES_GRACETIME=240000"
        ],
        "Cmd": null,
        "Image": "docker.io/homeassistant/home-assistant:latest",
        "Volumes": null,
        "WorkingDir": "/config",
        "Entrypoint": [
            "/init"
        ],
        "MacAddress": "02:42:c0:a8:01:1a",
        "OnBuild": null,
        "Labels": {
            "com.centurylinklabs.watchtower.depends-on": "zigbee2mqtt,mariadb",
            "com.centurylinklabs.watchtower.enable": "true",
            "io.hass.arch": "amd64",
            "io.hass.base.arch": "amd64",
            "io.hass.base.image": "ghcr.io/home-assistant/amd64-base:3.18",
            "io.hass.base.name": "python",
            "io.hass.base.version": "2023.10.4",
            "io.hass.type": "core",
            "io.hass.version": "2024.1.5",
            "org.opencontainers.image.authors": "The Home Assistant Authors",
            "org.opencontainers.image.created": "2024-01-20 19:03:51+00:00",
            "org.opencontainers.image.description": "Open-source home automation platform running on Python 3",
            "org.opencontainers.image.documentation": "https://www.home-assistant.io/docs/",
            "org.opencontainers.image.licenses": "Apache License 2.0",
            "org.opencontainers.image.source": "https://github.com/home-assistant/core",
            "org.opencontainers.image.title": "Home Assistant",
            "org.opencontainers.image.url": "https://www.home-assistant.io/",
            "org.opencontainers.image.version": "2024.1.5",
            "traefik.enable": "true",
            "traefik.http.routers.home-assistant.entrypoints": "websecure",
            "traefik.http.routers.home-assistant.rule": "Host(`home-assistant-web.home.insanegenius.net`)",
            "traefik.http.services.home-assistant.loadbalancer.server.port": "8123",
            "traefik.http.services.home-assistant.loadbalancer.server.scheme": "http"
        }
    },
    "NetworkSettings": {
        "Bridge": "",
        "SandboxID": "d90a7ee196f3927a19130bfbceef35ee172c3afbc45d00819e9de0e098fc1602",
        "SandboxKey": "/var/run/docker/netns/d90a7ee196f3",
        "Ports": {},
        "HairpinMode": false,
        "LinkLocalIPv6Address": "",
        "LinkLocalIPv6PrefixLen": 0,
        "SecondaryIPAddresses": null,
        "SecondaryIPv6Addresses": null,
        "EndpointID": "",
        "Gateway": "",
        "GlobalIPv6Address": "",
        "GlobalIPv6PrefixLen": 0,
        "IPAddress": "",
        "IPPrefixLen": 0,
        "IPv6Gateway": "",
        "MacAddress": "",
        "Networks": {
            "aaa_publicnet": {
                "IPAMConfig": null,
                "Links": null,
                "Aliases": [
                    "de6c479604e6",
                    "home-assistant"
                ],
                "MacAddress": "",
                "NetworkID": "233f37ba541b01a98137b9ff90137c409ca1b7c544b669e4d064448a3b66ea37",
                "EndpointID": "",
                "Gateway": "",
                "IPAddress": "",
                "IPPrefixLen": 0,
                "IPv6Gateway": "",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "DriverOpts": null,
                "DNSNames": [
                    "home-assistant",
                    "de6c479604e6"
                ]
            },
            "localnet": {
                "IPAMConfig": null,
                "Links": null,
                "Aliases": [
                    "ef0558eceda1",
                    "home-assistant",
                    "24b7994a80da",
                    "20e29aa7c68f",
                    "eb60c05ee6e1",
                    "60f02ac0e437",
                    "0738d3825b9a",
                    "c783e867f5ac",
                    "0d868a314a75",
                    "9dde6d93d4ff",
                    "dc70c3b10558",
                    "4a2685d09c65",
                    "6dcfac5b2253",
                    "646d0717fcde",
                    "6ef9b6d9ce50",
                    "de6c479604e6"
                ],
                "MacAddress": "",
                "NetworkID": "742700da91d804fe03f5b793f35309def9ee34e7f7a2c815eaa303cc32d4364d",
                "EndpointID": "",
                "Gateway": "",
                "IPAddress": "",
                "IPPrefixLen": 0,
                "IPv6Gateway": "",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "DriverOpts": null,
                "DNSNames": [
                    "home-assistant",
                    "ef0558eceda1",
                    "24b7994a80da",
                    "20e29aa7c68f",
                    "eb60c05ee6e1",
                    "60f02ac0e437",
                    "0738d3825b9a",
                    "c783e867f5ac",
                    "0d868a314a75",
                    "9dde6d93d4ff",
                    "dc70c3b10558",
                    "4a2685d09c65",
                    "6dcfac5b2253",
                    "646d0717fcde",
                    "6ef9b6d9ce50",
                    "de6c479604e6"
                ]
            }
        }
    }
}
```

Notice the difference in network config, no `IPAddress`:

```json
            "aaa_publicnet": {
                "IPAMConfig": {
                    "IPv4Address": "192.168.1.26"
                },
                "Links": null,
                "Aliases": [
                    "cb17b2321ee3",
                    "home-assistant"
                ],
                "MacAddress": "02:42:c0:a8:01:1a",
                "NetworkID": "233f37ba541b01a98137b9ff90137c409ca1b7c544b669e4d064448a3b66ea37",
                "EndpointID": "c56a194786f41f7ca5acea62d8993a73b7cb0caee3c94aeb680ac4b28bdf43da",
                "Gateway": "192.168.1.1",
                "IPAddress": "192.168.1.26",
                "IPPrefixLen": 24,
                "IPv6Gateway": "",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "DriverOpts": null,
                "DNSNames": [
                    "home-assistant",
                    "cb17b2321ee3"
                ]
            },
```

```json
            "aaa_publicnet": {
                "IPAMConfig": null,
                "Links": null,
                "Aliases": [
                    "de6c479604e6",
                    "home-assistant"
                ],
                "MacAddress": "",
                "NetworkID": "233f37ba541b01a98137b9ff90137c409ca1b7c544b669e4d064448a3b66ea37",
                "EndpointID": "",
                "Gateway": "",
                "IPAddress": "",
                "IPPrefixLen": 0,
                "IPv6Gateway": "",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "DriverOpts": null,
                "DNSNames": [
                    "home-assistant",
                    "de6c479604e6"
                ]
            },
```

**Comment by ptr727** [comment]
Seems to be fixed after 25.0.2, I had to recreate the containers else updates by watchtower kept failing.

**Comment by ptr727** [comment]
It happened again using docker 25.0.3, this time a different container, but again after WT recreated the container it was using the wrong static IP.

See discussion [here](https://github.com/moby/moby/discussions/47257) for details.

Looks like the recreated container has a `null` `IPAMConfig` and sets the IP as an attribute, while the original/correct version has the IP as part of `IPAMConfig`, that is now `null` after recreate.

**Comment by edrikk** [comment]
Hi,

@piksel I’m sorry if you aren’t the correct person to @ - It’s my first issue in this repository and noted that you are a key/top commiter.

With respect to this issue, robmry from Docker has done some deep analysis, and it appears as though the issue may require updates in WatchTower.

Please see this ticket in the docker git:  https://github.com/moby/moby/issues/47441#issuecomment-1969590931

As it stands Watchtower is require full manual effort after failed updates.  Would really appreciate if it could be looked at.

**Comment by NonaSuomy** [comment]
Would this fix watchtower not applying the mac_address? https://github.com/moby/moby/pull/47474#issuecomment-1974931738

**Comment by ptr727** [comment]
Looks to be resolved after Moby 25.04+.
